app/controllers/queue_items_controller.rb

```diff
@@ -14,6 +14,19 @@ def create
   def destroy
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if current_user.owner?(queue_item)
+    current_user.normalize_queue_item_position 
+    redirect_to my_queue_path
+  end
+
+  def update_queue
+    begin
+      update_queue_items
+    rescue ActiveRecord::RecordInvalid
+      flash[:error] = "Invalid position number"
+      redirect_to my_queue_path
+      return
+    end
+    current_user.normalize_queue_item_position 
     redirect_to my_queue_path
```

Comment:
HOMEWORK: refactor this method - you are redirecting to my_queue_path twice and using explicit return which is not needed. This method could be 2 lines leaner.